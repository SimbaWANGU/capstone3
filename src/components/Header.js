import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import '../styles/Header.scss';
import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';
import KeyboardVoiceIcon from '@mui/icons-material/KeyboardVoice';
import SettingsIcon from '@mui/icons-material/Settings';

function Header({ title }) {
  return (
    <div className="header">
      <NavLink to="/">
        <ArrowBackIosNewIcon className="back" style={{ fill: '#FAFAFA' }} />
      </NavLink>
      <p>2021</p>
      <h5>{title}</h5>
      <KeyboardVoiceIcon style={{ fill: '#FAFAFA' }} />
      <SettingsIcon style={{ fill: '#FAFAFA' }} />
    </div>
  );
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Header;
