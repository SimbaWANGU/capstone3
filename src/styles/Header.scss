$headerColor : rgba(67, 105, 178, 1);
$fontColor : rgba(255, 255, 255, 0.8);

.header {
  background-color: $headerColor;
  padding: 10px;
  display: flex;
  flex-direction: row;

  p {
    margin: auto auto auto 0;
    font-size: 18px;
    color: $fontColor;
    font-weight: 500;
  }

  h5 {
    margin: auto auto auto 0;
    font-size: 18px;
    color: $fontColor;
    font-weight: 500;
  }
}
