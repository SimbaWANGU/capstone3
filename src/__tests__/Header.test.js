import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Header from '../components/Header';

it('Renders correctly', () => {
  const component = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Header title="testing" />
      </BrowserRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
