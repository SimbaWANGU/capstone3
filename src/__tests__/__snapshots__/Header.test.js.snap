// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="header"
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium back css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowBackIosNewIcon"
      focusable="false"
      style={
        Object {
          "fill": "#FAFAFA",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M17.77 3.77 16 2 6 12l10 10 1.77-1.77L9.54 12z"
      />
    </svg>
  </a>
  <p>
    2021
  </p>
  <h5>
    testing
  </h5>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="KeyboardVoiceIcon"
    focusable="false"
    style={
      Object {
        "fill": "#FAFAFA",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 15c1.66 0 2.99-1.34 2.99-3L15 6c0-1.66-1.34-3-3-3S9 4.34 9 6v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 15 6.7 12H5c0 3.42 2.72 6.23 6 6.72V22h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
    />
  </svg>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="SettingsIcon"
    focusable="false"
    style={
      Object {
        "fill": "#FAFAFA",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
    />
  </svg>
</div>
`;
